File: backup/Search/python/11_카드정렬하기.py
# fast campus 강의
# https://www.acmicpc.net/problem/1715
# 2

import heapq

N = int(input())
arr = []
for _ in range(N):
    heapq.heappush(arr, int(input()))

result = 0
while len(arr) > 1:
    a = heapq.heappop(arr)
    b = heapq.heappop(arr)
    result += a + b
    heapq.heappush(arr, a + b)
print(result)

File: backup/Search/c++/11_카드정렬하기.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/1715
// 2

#include <iostream>
#include <vector>
#include <queue>

using namespace std;

int N;
priority_queue<int, vector<int>, greater<int>> pq;

int main() {
	cin >> N;
	for (int i = 0; i < N; i++) {
		int tmp;
		cin >> tmp;
		pq.push(tmp);
	}

	int result = 0;
	while (pq.size() > 1) {
		int a = pq.top(); pq.pop();
		int b = pq.top(); pq.pop();
		result += a + b;
		pq.push(a + b);
	}
	cout << result;
}